src/util/async/impl/ErasedOperation.hpp

```diff
@@ -119,22 +106,15 @@ class ErasedOperation {
         }
 
         void
-        requestStop() override
-        {
-            if constexpr (SomeStoppableOperation<OpType>) {
-                operation.requestStop();
-            } else {
-                ASSERT(false, "Stop requested on non-stoppable operation");
-            }
-        }
-
-        void
-        cancel() override
+        abort() override
         {
-            if constexpr (SomeCancellableOperation<OpType>) {
-                operation.cancel();
+            if constexpr (not SomeCancellableOperation<OpType> and not SomeStoppableOperation<OpType>) {
+                ASSERT(false, "Called abort() on an operation that can't be cancelled nor stopped");
```

Comment:
Maybe you could move it to concept? Looks like it could be checked at compile time.

Reply:
This can't be checked by a concept afaik. It's because this code is inside of a virtual function override.

Reply:
Maybe it is possible to put concept to `ErasedOperation::abort()` and add `static_assert()` here?

Reply:
We can't put a concept requirement